packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1426,6 +1463,85 @@ export class TransactionController extends BaseController<
 
     return dappSuggestedGasFees;
   }
+
+  /**
+   * Validates and adds external provided transaction to state.
+   *
+   * @param transactionMeta - Nominated external transaction to be added to state.
+   */
+  private async addExternalTransaction(transactionMeta: TransactionMeta) {
+    const { networkId, chainId } = this.getChainAndNetworkId();
+    const { transactions } = this.state;
+    const fromAddress = transactionMeta?.transaction?.from;
+    const sameFromAndNetworkTransactions = transactions.filter(
+      (transaction) =>
+        transaction.transaction.from === fromAddress &&
+        transactionMatchesNetwork(transaction, chainId, networkId),
+    );
+    const confirmedTxs = sameFromAndNetworkTransactions.filter(
+      (transaction) => transaction.status === TransactionStatus.confirmed,
+    );
+    const pendingTxs = this.state.transactions.filter(
+      (transaction) => transaction.status === TransactionStatus.submitted,
+    );
+
+    validateConfirmedExternalTransaction(
+      transactionMeta,
+      confirmedTxs,
+      pendingTxs,
+    );
+
+    const updatedTransactions = [...transactions, transactionMeta];
+    this.update({
+      transactions: this.trimTransactionsForState(updatedTransactions),
+    });
+  }
+
+  /**
+   * Sets other txMeta statuses to dropped if the txMeta that has been confirmed has other transactions
+   * in the transactions have the same nonce.
+   *
+   * @param transactionId - Used to identify original transaction.
+   */
+  private markNonceDuplicatesDropped(transactionId: string) {
+    const { networkId, chainId } = this.getChainAndNetworkId();
+    const transactionMeta = this.getTransaction(transactionId);
+    const nonce = transactionMeta?.transaction?.nonce;
+    const from = transactionMeta?.transaction?.from;
+    const sameNonceTxs = this.state.transactions.filter(
+      (transaction) =>
+        transaction.transaction.from === from &&
+        transaction.transaction.nonce === nonce &&
+        transactionMatchesNetwork(transaction, chainId, networkId),
+    );
+
+    if (!sameNonceTxs.length) {
+      return;
+    }
+
+    // Mark all same nonce transactions as dropped and give it a replacedBy hash
+    for (const transaction of sameNonceTxs) {
+      if (transaction.id === transactionId) {
+        return;
```

Comment:
As we're not a callback, this should now be `continue`.

Reply:
Done in https://github.com/MetaMask/core/pull/1625/commits/70ce5ad7ffa3e006f353b55f301c72f933f4e5af